src/OpenSage.Game/Terrain/Roads/CrossingRoadSegment.cs

```diff
@@ -0,0 +1,341 @@
+            var lengthToLeft = halfRoadWidth;
+            var lengthToRight = targetSize.Width - lengthToLeft;
+            var lengthToSide = 1.047f * template.RoadWidth;
+
+            var topPosition = crossingPosition + lengthToTop * upDirection;
+            var bottomPosition = crossingPosition - lengthToBottom * upDirection;
+            var sidePosition = crossingPosition + lengthToSide * sideDirection;
+
+            var midHeightOnMainRoad = 0.5f * (topPosition + bottomPosition);
+            var start = midHeightOnMainRoad - lengthToLeft * rightDirection;
+            var end = midHeightOnMainRoad + lengthToRight * rightDirection;
+
+            var top = new RoadSegmentEndPoint(topPosition);
+            var bottom = new RoadSegmentEndPoint(bottomPosition);
+            var side = new RoadSegmentEndPoint(sidePosition);
+
+            var crossingSegment = new CrossingRoadSegment(
+                crossingPosition,
+                new[] { top, bottom, side },
+                start,
+                end,
+                RoadTextureType.AsymmetricYCrossing);
+
+            var topEdge = mirror ? maxAngle.TopologyEdge : maxAngle.Previous.TopologyEdge;
+            var bottomEdge = mirror ? maxAngle.Previous.TopologyEdge : maxAngle.TopologyEdge;
+
+            Connect(crossingSegment, topEdge, top, upDirection, 0.04f * template.RoadWidth, edgeSegments);
+            Connect(crossingSegment, maxAngle.Previous.Previous.TopologyEdge, side, sideDirection, 0, edgeSegments);
+            Connect(crossingSegment, bottomEdge, bottom, -upDirection, 0.055f * template.RoadWidth, edgeSegments);
+
+            crossingSegment.MirrorTexture = mirror;
+
+            return crossingSegment;
+        }
+
+        private static CrossingRoadSegment CreateYSymmCrossing(
+           IEnumerable<IncomingRoadData> roads,
+           Vector3 crossingPosition,
+           RoadTemplate template,
+           IReadOnlyDictionary<RoadTopologyEdge, StraightRoadSegment> edgeSegments)
+        {
+            var minAngle = roads.OrderBy(road => road.AngleToPreviousEdge).First();
+            var mirror = false;
+
+            var rightEdge = minAngle;
+            var leftEdge = minAngle.Previous;
+            var topEdge = leftEdge.Previous;
+
+            var upDirection = topEdge.Direction;
+            var rightDirection = Vector3.Cross(upDirection, Vector3.UnitZ);
+            var rightSideDirection = Vector3.Normalize(rightDirection - upDirection);
+            var leftSideDirection = Vector3.Normalize(-rightDirection - upDirection);
+
+            var targetSize = GetSize(RoadTextureType.SymmetricYCrossing, template);
+
+            var lengthToTop = 0.269f * targetSize.Height;
+            var lengthToBottom = targetSize.Height - lengthToTop;
+
+            var halfRoadWidth = template.RoadWidth * template.RoadWidthInTexture / 2f;
+            var lengthToSide = 0.547f * template.RoadWidth;
```

Comment:
Do all these magic constants have some mathematical background or are they just defined like this by the engine?

Reply:
No idea to be honest...we tried to deduce and name as much as possible, but some magic numbers are left.